subsys/bluetooth/host/gatt.c

```diff
@@ -1990,7 +1990,7 @@ static int gatt_notify(struct bt_conn *conn, uint16_t handle,
 		int err;
 
 		err = gatt_notify_mult(conn, handle, params);
-		if (err && err != -ENOMEM) {
+		if (err != -ENOMEM) {
```

Comment:
Why is ENOMEM treated specially and the code is allowed to continue until the end of this function?

Reply:
I thought it was done intentionally in case of buf allocation failed for multiple notify PDU which needs a bit more memory than regular notify. if fails with ENOMEM and success with regular notify below, it would differ:
- opcode would be 0x1b instead of 0x23
- notification pdu would not include "len" information

As side note, I see another issue that we always send one handle per BT_ATT_OP_NOTIFY_MULT which add supports for  putting different (Handle, Len, Value) in same PDU, so it make it more effective in such case that multiple characteristic needs to be notified in regular intervals. I will follow and discuss this further

Reply:
Let's just do this the simple way: `if(err) {`. I don't think it's worth retrying the allocation as they will allocate from the same pool.

Reply:
@carlescufi actually even simpler: just `return gatt_notify_mult(conn, handle, params);` i.e. `err` goes away